src/category_theory/triangulated/rotate.lean

```diff
@@ -0,0 +1,629 @@
+```
+applying "rotate" gives a triangle morphism of the form:
+```
+      g        h       -f[1]
+  Y  ---> Z  --->  X[1] ---> Y[1]
+  |       |         |         |
+  |b      |c        |a[1]     |b[1]
+  V       V         V         V
+  Y' ---> Z' ---> X'[1] ---> Y'[1]
+      g'      h'       -f'[1]
+```
+-/
+@[simps]
+def rotate (f : triangle_morphism T₁ T₂)
+: triangle_morphism (T₁.rotate C) (T₂.rotate C):=
+{ trimor1 := f.trimor2,
+  trimor2 := f.trimor3,
+  trimor3 := f.trimor1⟦1⟧',
+  comm1 := by exact f.comm2,
+  comm2 := by exact f.comm3,
+  comm3 := begin
+    repeat {rw triangle.rotate_mor3},
+    rw [comp_neg, neg_comp],
+    repeat {rw ← functor.map_comp},
+    rw f.comm1,
+  end }
+
+/--
+Given a triangle morphism of the form:
+```
+      f       g       h
+  X  ---> Y  ---> Z  ---> X[1]
+  |       |       |        |
+  |a      |b      |c       |a[1]
+  V       V       V        V
+  X' ---> Y' ---> Z' ---> X'[1]
+      f'      g'      h'
+```
+applying "inv_rotate" gives a triangle morphism that can be thought of as:
+```
+        -h[-1]      f         g
+  Z[-1]  --->  X   --->  Y   --->  Z
+    |          |         |         |
+    |a         |b        |c        |a[1]
+    V          V         V         V
+  Z'[-1] --->  X'  --->  Y'  --->  Z'
+        -h'[-1]     f'        g'
+```
+(note that this diagram doesn't technically fit the definition of triangle morphism,
+as `Z[-1][1]` is not necessarily equal to `Z`, and `Z'[-1][1]` is not necessarily equal to `Z'`,
+but they are isomorphic, by the counit_iso of (shift C))
+-/
+@[simps]
+def inv_rotate (f : triangle_morphism T₁ T₂)
+: triangle_morphism (T₁.inv_rotate C) (T₂.inv_rotate C) :=
+{ trimor1 := f.trimor3⟦-1⟧',
+  trimor2 := f.trimor1,
+  trimor3 := f.trimor2,
+  comm1 := begin
+    simp only [triangle.inv_rotate_mor1],
+    rw [comp_neg, neg_comp, ← assoc],
+    dsimp,
+    rw [← functor.map_comp (shift C ).inverse, ← f.comm3, functor.map_comp],
+    repeat {rw assoc},
+    suffices h : (shift C).unit_iso.inv.app T₁.obj1 ≫ f.trimor1 =
+    (shift C).inverse.map ((shift C).functor.map f.trimor1) ≫ (shift C).unit_iso.inv.app T₂.obj1,
+    {
+      rw h,
+      refl,
+    },
+    {
+      simp only [iso.hom_inv_id_app, assoc, equivalence.inv_fun_map,
+      nat_iso.cancel_nat_iso_inv_left],
+      exact (category.comp_id f.trimor1).symm,
+    }
+  end,
+  comm2 := by exact f.comm1,
+  comm3 := begin
+    have h := f.comm2,
+    repeat {rw triangle.inv_rotate_mor3},
+    rw [← assoc f.trimor2 _, ← f.comm2],
+    dsimp,
+    repeat {rw assoc},
+    simp only [equivalence.fun_inv_map, iso.inv_hom_id_app_assoc],
+  end }
+
+end triangle_morphism
+
+/--
+Rotating triangles gives an endofunctor on the category of triangles in C.
+-/
+@[simps]
+def rotate : (triangle C) ⥤ (triangle C) :=
+{ obj := triangle.rotate C,
+  map := λ _ _ f, f.rotate C,
+  map_id' := begin
+    intro T₁,
+    simp only [triangle_category_to_category_struct_id],
+    unfold triangle_morphism.rotate,
```

Comment:
Hmm, why is `unfold` necessary? This is usually a code smell that `@[simp]` lemmas are missing.